MoltenVK/MoltenVK/GPUObjects/MVKPipeline.mm

```diff
@@ -2348,7 +2244,8 @@ void serialize(Archive & archive, CompilerMSL::Options& opt) {
 				opt.ios_use_simdgroup_functions,
 				opt.emulate_subgroups,
 				opt.vertex_index_type,
-				opt.force_sample_rate_shading);
+				opt.force_sample_rate_shading,
+				opt.raw_buffer_tese_input);
 	}
```

Comment:
Looks like this Archive struct is missing the new `shader_patch_input_buffer_index` member. This is causing the pipeline cache values being read in to never match what was stored to them whenever this member is set to anything other than the default value. This affects both the tess stage and the following frag stage, since they use the same `SPIRVToMSLConversionConfiguration` struct.

Can you add `shader_patch_input_buffer_index` in the appropriate place in Archive definition, between these two existing entries, please?

    opt.shader_index_buffer_index,
    opt.shader_patch_input_buffer_index,     //  <--new
    opt.shader_input_wg_index,

Reply:
/me smacks head

D'oh! Fixed.